src/ORM/Association/Loader/SelectLoader.php

```diff
@@ -9,40 +9,115 @@
  *
  * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
  * @link          http://cakephp.org CakePHP(tm) Project
- * @since         3.0.0
+ * @since         3.4.0
  * @license       http://www.opensource.org/licenses/mit-license.php MIT License
  */
-namespace Cake\ORM\Association;
+namespace Cake\ORM\Association\Loader;
 
 use Cake\Database\Expression\IdentifierExpression;
 use Cake\Database\Expression\TupleComparison;
 use Cake\Database\ValueBinder;
+use Cake\ORM\Association;
 use InvalidArgumentException;
+use RuntimeException;
 
 /**
- * Represents a type of association that that can be fetched using another query
+ * Implements the logic for loading an association using a SELECT query
  */
```

Comment:
Should this class be marked as `@internal` or is it part of the 'public' API for the ORM?